PaintingTrivia.py

```diff
@@ -18,25 +32,77 @@ def set_session():
 def index():
     if 'username' in session:
         # return 'Logged in as %s' % Markup.escape(session['username'])
-        print 'index to show entries'
         return redirect(url_for('show_entries'))
     else:
-        print 'index to login'
-        return redirect(url_for('login'))
+        return redirect(url_for('menu'))
+
+
+@app.route('/main', methods=['GET', 'POST'])
+def menu():
+    if request.method == 'POST':
+        print 'Hey'
+        x = layout_buttons(request)
+        if x:
+            return x
+    return render_template('menu.html')
 
 
 @app.route('/login', methods=['GET', 'POST'])
 def login():
     if request.method == 'POST':
-        session['username'] = request.form['username']
-        return redirect(url_for('index'))
-    return render_template('menu.html')
+        x = layout_buttons(request)
+        if x:
+            return x
+        username_aux = request.form['username']
+        password_aux = request.form['password']
+        if not (username_aux == '' or password_aux == ''):
+            # Check if the username exists
+            retrieved_users = User.query.filter(User.name == username_aux).order_by(func.random()).all()
+            if retrieved_users:
+                # Check if the password is correct
+                if retrieved_users[0].password == password_aux:
+                    session['username'] = username_aux
+                    session['password'] = password_aux
+                    return redirect(url_for('index'))
+                else:
+                    flash('Password is wrong!', 'error')
+                    return redirect(url_for('index'))
+            else:
+                flash('This user is not registered!', 'error')
+                return redirect(url_for('register'))
+    return render_template('login.html')
+
+
+@app.route('/register', methods=['GET', 'POST'])
+def register():
+    if request.method == 'POST':
+        x = layout_buttons(request)
+        if x:
+            return x
+        username_aux = request.form['username']
+        password_aux = request.form['password']
+        if not (username_aux == '' or password_aux == ''):
+            session['username'] = username_aux
+            session['password'] = password_aux
+            retrieved_users = User.query.filter(User.name == username_aux).order_by(func.random()).all()
+            if retrieved_users:
+                # Check if the password is correct
+                flash('User already exists', 'error')
+            else:
+                # Add the user
+                session['username'] = username_aux
+                session['password'] = password_aux
+                users.add_user(username_aux, password_aux)
```

Comment:
you are saving password in clear text (bad). see http://stackoverflow.com/questions/6058019/storing-user-and-password-in-a-database